src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -143,13 +143,39 @@ class UnityPlugin implements Plugin<Project> {
     }
 
     private static void configureUnityTasks(UnityPluginExtension extension, final Project project) {
+
+        // Generate the environment/gradle.properties providers
+        Map<UnityCommandLineOption, Provider<String>> stringOptionProviders =
+            new HashMap<UnityCommandLineOption, Provider<String>>()
+        Map<UnityCommandLineOption, Provider<Boolean>> booleanOptionProviders =
+            new HashMap<UnityCommandLineOption, Provider<Boolean>>()
+        for (option in UnityCommandLineOption.values()) {
+            if (!option.map) {
+                continue
+            }
+            def propertyKey = "unity.${option}"
```

Comment:
the hardcoded "unity" should be the extension name. We prefix the property value and env variable with the extension name so that it becomes clear how to name env and property values.

extension = foo
property = bar

in the script access could be:
```groovy
foo {
    bar = "baz"
}
```
or
```groovy
foo.bar="baz"
```

The last example demonstrates how the gradle property looks like it looks like.

For env it is a simple conversion of camelcase to upper snake case and replaceing all dots with `_`.

Reply:
Yes, will change that.

Reply:
you did not :)

Reply:
Argh, done now.